core/src/test/scala/scala/spores/run/basic/Basic.scala

```diff
@@ -23,11 +23,180 @@ package somepackage {
   }
 }
 
+
+
+abstract class X { def g(f: Int => Unit) : Unit}
+abstract class TestCl[T] {val x : X}
+
+
 @RunWith(classOf[JUnit4])
 class BasicSpec {
+
+
+  /**
+    * Test the following:
+    * spore has variable a
+    * a is part of PTT in spore
+    * PTT needs to be correct, when 'a' is changed into x$macro$n,
+    * PTT must change.
+    *
+    * Expansion with SporeConv.sporeConv is optional.
+    */
+  @Test
+  def pTTParameterCapture(): Unit = {
+
+    abstract class A {
+      self =>
+      type B
+      val t: self.B
+      def f(x: self.B) = {}
+    }
+
+    val s = spore{
+      val y = 5
+      (a: A) => {
+        val k: a.B = a.t
+        a.f(k)
+      }
+    }
+  }
+
+  /**
+    * Same as pTTParameterCapture, but
+    * the path-dependent type comes from a captured variable,
+    * not from the parameter
+    */
+  @Test
+  def pTTCapturedCapture(): Unit = {
+
+    abstract class A {
+      self =>
+      type B
+      val t: self.B
+      def f(x: self.B) = {}
+    }
+
+    val s = spore{
+      (a: A) => {
+        val s2 = spore {
+          val na = a
+          (_: Unit) => {
+            val k: na.B = na.t
+            na.f(k)
+          }
+        }
+      }
+    }
+  }
+
+  /**
+    * Same as pTTCapturedCapture, but with nullary spores (more code coverage)
+    */
+  @Test
+  def pTTCapturedCaptureNullary(): Unit = {
+
+    abstract class A {
+      self =>
+      type B
+      val t: self.B
+      def f(x: self.B) = {}
+    }
+
+    val s = spore{
+      (a: A) => {
+        val s2 = spore {
+          val na = a
+          delayed {
+            val k: na.B = na.t
+            na.f(k)
+          }
+        }
+      }
+    }
+  }
+
+  /**
+    * Similar to pTTCapture(), but spores are nested.
+    */
+  @Test
+  def nestedPTTCapture(): Unit = {
+
+    abstract class A {
+      self =>
+      type B
+      val t: self.B
+      def f(x: self.B) = {}
+    }
+
+    val s = spore{
+      val y = 5
+      (_: Unit) => {
+        spore {
+          val y = 5
+          (a: A) => {
+            val k: a.B = a.t
+            a.f(k)
+          }
+        }
+        println()
+      }
+    }
+  }
+
+
+  /**
+    * same as nestedPTTCapture, but (_: Unit) => ... replaced with Nullary spore
+    */
+//  @Test
+//  def nestedPTTCaptureNullary(): Unit = {
+//
+//    abstract class A {
+//      self =>
+//      type B
+//      val t: self.B
+//      def f(x: self.B) = {}
+//    }
+//
+//    val s = spore{
+//      val y = 5
+//      delayed {
+//        spore {
+//          val y = 5
+//          (a: A) => {
+//            val k: a.B = a.t
+//            a.f(k)
+//          }
+//        }
+//        println()
+//      }
+//    }
+//  }
+
+
+
+  /**
+    * Should be the same that SporeAvoided and Spore expands to in Ownership.scala.
```

Comment:
Probably better to remove that comment, since it refers to things outside of this repo.